src/array/binary/array.rs

```diff
@@ -69,6 +76,13 @@ impl<'a, O: OffsetSizeTrait> GeometryArrayTrait<'a> for WKBArray<O> {
             "ARROW:extension:name".to_string(),
             self.extension_name().to_string(),
         );
+        metadata.insert(
+            "flavor".to_string(),
+            match self.flavor {
+                WKBFlavor::ISO => "iso".to_string(),
+                WKBFlavor::EWKB => "ewkb".to_string(),
+            },
+        );
```

Comment:
Here you're inserting into the wrong metadata hashmap. Geoarrow has its _own_ hashmap as json inside the key `"ARROW:extension:metadata"`